### PR TITLE
Update configuration to generate docs from docstrings

### PR DESCRIPTION
**Pull request by Andres-GC**
This PR updates the configuration to generate the HTML files for the documentation from the code's docstrings automatically.
When `make docs` is run, it will clean the current version of the files and generate new ones. We might want to change this behavior in the future when we have an stable version of the library.

The PR includes:
- Updates on the make task
- All the autogenerated `.rst` files under `docs/source/` directory.
- Updates on the docstring to follow the conventions listed on [CONTRIBUTING.md](https://github.com/HewlettPackard/py-spiffe/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/pyspiffe/spiffe_id/trust_domain.py [hunk 1]
```diff
@@ -9,25 +14,26 @@
...
 
-    Args:
-        name(str): The name of the Trust Domain
+    def __init__(self, name: str) -> None:
+        """Creates a new TrustDomain Object.
 
-    Raises:
-        ValueError: if the name of the trust domain is empty, has a port, or contains an invalid scheme.
+        Args:
+            name: The name of the Trust Domain
```
Comment (glaucimar) [inline src/pyspiffe/spiffe_id/trust_domain.py:23]: nit: missing a dot.

File: src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py [hunk 2]
```diff
@@ -1,28 +1,43 @@
...
-    ) -> Mapping[TrustDomain, JwtBundle]:
-        """
-        NewSet creates a new, empty set.
-        """
-        raise Exception('not implemented')
+    def __init__(self, bundles: Mapping[TrustDomain, JwtBundle]) -> None:
+        """Creates a new initialized with the given JWT bundles.
+
+        Args:
+            bundles: a set of JwtBundles to initialize the JwtBundleSet
```
Comment (glaucimar) [inline src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py:17]: Missing dot.

File: src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
-        """
-        returns the JWT bundle of the given trust domain.
+    def get_jwt_bundle_for_trust_domain(self, trust_domain: TrustDomain) -> JwtBundle:
+        """Returns the JWT bundle of the given trust domain.
+
+        Args:
+            trust_domain: The TrustDomain to get a JwtBundle.
+
+        Returns:
+            a JwtBundle for the given TrustDomain
```
Comment (glaucimar) [inline src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py:41]: nit: missing a dot at the end of the line.

File: src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py [hunk 4]
```diff
@@ -1,28 +1,43 @@
...
+        pass
+
+    def add(self, jwt_bundle: JwtBundle):
+        """Adds a new bundle into the set.
+
+        If a bundle already exists for the trust domain, the existing bundle is
+        replaced.
+
+        Args:
+            jwt_bundle: The new JwtBundle to add
```
Comment (glaucimar) [inline src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py:30]: nit1: missing a dot at the end. Other occurrences of these in this PR ;).

File: src/pyspiffe/bundle/jwt_bundle/jwt_bundle.py [hunk 5]
```diff
@@ -1,2 +1,9 @@
+"""
+JwtBundle module manage JwtBundle objects.
+"""
```
Comment (glaucimar) [inline src/pyspiffe/bundle/jwt_bundle/jwt_bundle.py:3]: ..module manages...
Couple of other occurrences of this.

File: src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py [hunk 6]
```diff
@@ -1,28 +1,43 @@
...
         """
         raise Exception('not implemented')
```
Comment (glaucimar) [inline src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py:32]: error messages should also end with a dot. I understand this is not from this PR but.. if you are willing to make the change. :)

File: src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py (hunk 2 above)
Comment (glaucimar) [inline src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py:17]: nit: after semicolom, use capital letter. So, it should be:
`bundles: A set of Jwt`
Please, review the other files for this one.

File: src/pyspiffe/spiffe_id/__init__.py [hunk 7]
```diff
@@ -1 +1,6 @@
+"""
+spiffe_id Module. Contains information related to SPIFFE Ids
```
Comment (glaucimar) [inline src/pyspiffe/spiffe_id/__init__.py:2]: nit: missing dot at the end of the line.

File: src/pyspiffe/spiffe_id/trust_domain.py [hunk 8]
```diff
@@ -1,6 +1,11 @@
...
+This module manage TrustDomain objects.
+"""
+
 from urllib.parse import urlparse, ParseResult
 from typing import Any, cast
 
+
 from pyspiffe.spiffe_id import SPIFFE_SCHEME
 
 EMPTY_DOMAIN_ERROR = 'Trust domain cannot be empty.'
```
Comment (glaucimar) [inline src/pyspiffe/spiffe_id/trust_domain.py:11]: missing docstring.

Comment (Andres-GC) [reply src/pyspiffe/spiffe_id/trust_domain.py:11]: Thanks for the comments Glaucimar.
I've covered all of them. I added docstrings for these module variables. At some point we should check whether they need to be public to users of the module or not.

**Review by scotte — APPROVED** [review]

**Review by maxlambrecht — APPROVED** [review]

**Pull request merged**